### PR TITLE
Use unique `ConfigMap`s/`Secret`s for Helm chart components

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area robustness
/kind enhancement

**What this PR does / why we need it**:
This PR adapts the Helm chart components to use unique `ConfigMap`s/`Secret`s (i.e., those with a unique name and with `.immutable=true`).

**Which issue(s) this PR fixes**:
Part of #3484

**Special notes for your reviewer:**
This is one of a two PRs for above mentioned issue. The first PR is handling the Golang components (#4518).

/squash

**Release note**:

```noteworthy operator
NONE
```

### CODE REVIEW

File: charts/seed-bootstrap/charts/nginx-ingress/templates/_helpers.tpl [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{{- define "nginx-ingress.config.data" -}}
+{{- range $k, $v := .Values.config }}
+{{ $k }}: {{ quote $v }}
+{{- end }}
+{{- end -}}
+
+{{- define "nginx-ingress.config.name" -}}
+nginx-ingress-controller-{{ include "nginx-ingress.config.data" . | sha256sum | trunc 8 }}
```
Comment (timuthy) [inline charts/seed-bootstrap/charts/nginx-ingress/templates/_helpers.tpl:8]: I'm wondering if this works when there is no config specified for the ingress controller?

**Review by timuthy — COMMENTED** [review]
Looks good to me so far, thank you. I just left one question.

File: charts/seed-bootstrap/charts/nginx-ingress/templates/_helpers.tpl (hunk 1 above)
Comment (rfranzke) [reply charts/seed-bootstrap/charts/nginx-ingress/templates/_helpers.tpl:8]: Yeah, it works well, you can try locally by setting `config: {}` in the Helm chart.

**Review by timuthy — APPROVED** [review]
/lgtm

**Pull request merged**